frontend/src/metabase/visualizations/lib/save-dashboard-pdf.ts

```diff
@@ -0,0 +1,69 @@
+import html2canvas from "html2canvas";
+import jspdf from "jspdf";
+
+import { color } from "metabase/lib/colors";
+
+import { SAVING_CHART_IMAGE_CLASS } from "./save-chart-image";
+
+export const saveDashboardPdf = async (
+  selector: string,
+  dashboardName: string,
+) => {
+  const fileName = `${dashboardName}.pdf`;
+  const node = document.querySelector(selector);
+
+  if (!node || !(node instanceof HTMLElement)) {
+    console.warn("No node found for selector", selector);
+    return;
+  }
+
+  node.classList.add(SAVING_CHART_IMAGE_CLASS);
+
+  const canvas = await html2canvas(node, {
+    useCORS: true,
+    onclone: (doc: Document, node: HTMLElement) => {
+      const style = doc.createElement("style");
+      style.innerHTML = `
```

Comment:
I think mixing two different approaches to tweaking styles while saving DOM is not the best approach.
1) Having a global class '.saving-chart-image' which is used in components to hide/adjust certain elements while saving the image.
2) Changing styles in the saving image code right away.

Initially, I picked the first approach because it leaves components to define how they are present while being saved as an image. So that when the next engineers edit them they see that components should be rendered in a certain when using this functionality. Otherwise, if we patch CSS only in this file, any further changes in the components that are being captured can break the functionality since it is not obvious they should be tweaked.

Wdyt about sticking to the first approach? Also, I think we could rename `SAVING_CHART_IMAGE_CLASS`, `saving-chart-image`, and other related code to reflect that it is not for charts only but for dom in general